src/platform/user/tests/authentication/downtime.unit.spec.js

```diff
@@ -0,0 +1,241 @@
+    expect(alert.getAttribute('status')).to.equal('warning');
+    expect(alert.querySelector('h2').textContent).to.equal(
+      'You may have trouble signing in or using some tools or services',
+    );
+  });
+
+  it('should render banner for a single service down', () => {
+    const statuses = [{ serviceId: 'logingov', status: 'down' }];
+    const { container } = render(downtimeUtils.renderDowntimeBanner(statuses));
+    const alert = container.querySelector('va-alert');
+    expect(alert).to.exist;
+    expect(alert.getAttribute('status')).to.equal('warning');
+    expect(alert.querySelector('h2').textContent).to.equal(
+      'You may have trouble signing in with Login.gov',
+    );
+  });
+
+  it('should return null when no services are down', () => {
+    const statuses = [{ serviceId: 'loginGov', status: 'active' }];
+    const result = downtimeUtils.renderDowntimeBanner(statuses);
+    expect(result).to.be.null;
+  });
+
+  it('should return null when provided an empty status list', () => {
+    const statuses = [];
+    const result = downtimeUtils.renderDowntimeBanner(statuses);
+    expect(result).to.be.null;
+  });
+});
+
+describe('createMaintenanceBanner', () => {
+  it('should correctly construct a maintenance banner object', () => {
+    const startTime = new Date().toISOString();
+    const endTime = new Date(
+      new Date().getTime() + 2 * 60 * 60 * 1000,
+    ).toISOString(); // 2 hours later
+    const banner = downtimeUtils.createMaintenanceBanner({
+      start_time: startTime,
+      end_time: endTime,
+    });
+
+    expect(banner).to.have.property('headline', 'Upcoming site maintenance');
+    expect(banner).to.have.property('status', 'info');
+    expect(banner)
+      .to.have.property('message')
+      .that.is.a('object');
+    expect(banner)
+      .to.have.property('startTime')
+      .that.is.a('date');
+    expect(banner)
+      .to.have.property('endTime')
+      .that.is.a('date');
+  });
+});
+
+describe('determineMaintenance', () => {
+  it('should find the first maintenance window that meets the criteria', () => {
+    const maintArray = [
+      { external_service: 'logingov' },
+      { external_service: 'global' },
+    ];
+    const result = downtimeUtils.determineMaintenance(maintArray);
+    expect(result).to.deep.equal({ external_service: 'logingov' });
+  });
+
+  it('should return undefined if no maintenance window meets the criteria', () => {
+    const maintArray = [{ external_service: 'otherService' }];
+    const result = downtimeUtils.determineMaintenance(maintArray);
+    expect(result).to.be.undefined;
+  });
+});
+
+describe('isInMaintenanceWindow', () => {
+  it('should return true when the current time is within the maintenance window', () => {
+    const startTime = new Date(
+      new Date().getTime() - 1 * 60 * 60 * 1000,
+    ).toISOString(); // 1 hour ago
+    const endTime = new Date(
+      new Date().getTime() + 1 * 60 * 60 * 1000,
+    ).toISOString(); // 1 hour later
```

Comment:
It might be worth creating some constants (or a helper function) for these repeat time calculations.